provider/datagen/src/transform/cldr/datetime/neo.rs

```diff
@@ -451,14 +482,36 @@ fn months_convert(
                 panic!("CLDR month indices cannot be zero");
             }
 
-            symbols[index - 1] = &v;
+            symbols[index - 1] = v.into();
         }
-        for (i, val) in symbols.iter().enumerate() {
+
+        let nonleap = if has_leap {
+            month_count / 2
+        } else {
+            month_count
+        };
+
+        for (i, val) in symbols.iter().take(nonleap).enumerate() {
             if val.is_empty() {
                 panic!("Calendar {calendar} does not have data for month {i}; found data for {symbols:?}");
             }
         }
-        Ok(MonthSymbolsV1::Numeric((&symbols).into()))
+
+        if has_leap {
```

Comment:
Thought: Somewhat confusing because this is really `if has_leap && calendar != "hebrew"` since we're inside a nested if statement